src/server/game/Warden/WardenWin.cpp

```diff
@@ -239,6 +252,80 @@ void WardenWin::HandleHashResult(ByteBuffer& buff)
     _initialized = true;
 }
 
+uint16 WardenWin::GetFreePayloadId()
+{
+    uint16 payloadId = _wardenPayloadOffset;
+
+    while (CachedChecks.find(payloadId) != CachedChecks.end())
```

Comment:
map.find returns an iterator you don't need the while here. Just use if.

Reply:
I am unsure of how I can use the iterator here to get a free id, could you give me some further advice?

Reply:
```while (CachedChecks.find(payloadId) != CachedChecks.end())```

To:

```if (CachedChecks.find(payloadId) != CachedChecks.end())```

Reply:
Wouldn't be correct in this case, he wants to find the next free ID to use for the payload, not necessarily an already queued payload

Reply:
If you want to you could probably just get the last inserted id and add + 1 to it.

Reply:
It's probably fine how it is currently, RegisterPayload is designed to be called once per static payload so it won't be that expensive. Thanks for reviewing this however. :)

Reply:
```c++
uint16 WardenWin::GetFreePayloadId()
{
	uint16 payloadId = _wardenPayloadOffset;

	auto it = CachedChecks.find(payloadId);
	if (it != CachedChecks.end())
	{
            payloadId = std::prev(CachedChecks.end())->first + 1;
	}

	return payloadId;
}

```

This would also require CachedChecks to be map not the unordered version.

Reply:
I'll give that a test but it looks ok.

Reply:
The reason I wanted it to be important that it finds a free payloadId is because I'm pretty sure the client expects a value with no more than a length of 4 (9999 max) so I was trying to re-use old ids that have been removed from the cache. If it's safe to assume the session won't cache 4999 payloads then this solution is fine. If you are caching 4999 warden payloads you probably have other issues anyway so I'll accept this change.